CodeStarterCamp_Week3/main.swift

```diff
@@ -9,33 +9,56 @@
 import Foundation
 
 class Person {
-    var money: Int = 0
+    var money: Int
     
-    func buyCoffee() {
-        print("buy coffee.")
+    init(money: Int) {
+        self.money = money
+    }
+    
+    func order(_ coffee: Coffee, of coffeeshop: CoffeeShop, by name: String) {
+        coffeeshop.make(coffee, for: name)
+        if let coffeePrice: Int = coffeeshop.menuTable[coffee] {
+            if money >= coffeePrice {
+                money -= coffeePrice
+            } else {
+                print("잔액이 \(coffeePrice - money)만큼 부족합니다.")
+            }
+            coffeeshop.sales += coffeePrice
+        }
     }
 }
 
 class CoffeeShop {
     var sales: Int = 0
-    var menuTable: [String: Int] = [:]
-    var pickUpTable: [Coffee] = []
-    var barista: Person?
-    
-    func takeOrder(order: Coffee) {
-        makeCoffee(coffee: order)
+    var menuTable: [Coffee: Int] = [
+        .americano: 4500,
+        .latte : 4500,
+        .espresso : 4000,
+        .cappuccino : 4500
+    ]
+    var pickUpTable: [String] = [] {
+        didSet {
+            if let newCoffee = pickUpTable.last, let customer = customer {
+                print("\(customer) 님이 주문하신 \(newCoffee)(이/가) 준비되었습니다. 픽업대에서 가져가주세요.")
+            }
+        }
     }
+    var barista: Person?
+    var customer: String?
     
-    func makeCoffee(coffee: Coffee) {
-        pickUpTable.append(coffee)
+    func make(_ coffee: Coffee, for name: String) {
+        customer = name
+        pickUpTable.append(coffee.rawValue)
     }
 }
 
-enum Coffee {
-    case americano, latte, espresso, iceTea
+enum Coffee: String {
+    case americano = "아메리카노", latte = "라떼", espresso = "에스프레소", cappuccino = "카푸치노"
 }
 
-let misterLee: Person = Person()
-let missKim: Person = Person()
+let misterLee: Person = Person(money: 0)
+let missKim: Person = Person(money: 3500)
 let yagombucks: CoffeeShop = CoffeeShop()
 yagombucks.barista = misterLee
+
+missKim.order(.americano, of: yagombucks, by: "missKim")
```

Comment:
Person 객체에 name 프로퍼티가 있다면!
```suggestion
missKim.order(.americano, of: yagombucks, by: missKim.name)
```
으로도 수정이 가능해서, 훨씬 직관적일 거 같네요!